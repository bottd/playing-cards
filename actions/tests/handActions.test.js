import {
  actionTypes,
  addCardToHand,
  discardCard,
  discardHand,
  hideCard,
  revealCard,
} from '../handActions';

describe('Hand actions', () => {
  describe('Action types', () => {
    it('Should match snapshot', () => {
      expect(actionTypes).toMatchSnapshot();
    });
  });

  describe('addCardToHand', () => {
    const mockCard = {
      id: 1,
      name: 'Ace of Spades',
    };
    it('Should have the correct action type', () => {
      const expected = 'ADD_CARD_TO_HAND';
      const result = addCardToHand(mockCard);
      expect(result.type).toEqual(expected);
    });
    it('Should return an action with passed card', () => {
      const result = addCardToHand(mockCard);
      expect(result.card).toEqual(mockCard);
    });
    it('Should match snapshot', () => {
      expect(addCardToHand(mockCard)).toMatchSnapshot();
    });
  });

  describe('discardCard', () => {
    const mockCard = {
      id: 1,
      name: 'Ace of Spades',
    };
    it('Should have the correct action type', () => {
      const expected = 'DISCARD_CARD';
      const result = discardCard(mockCard);
      expect(result.type).toBe(expected);
    });
    it('Should return an action with passed card', () => {
      const result = discardCard(mockCard);
      expect(result.card).toEqual(mockCard);
    });
    it('Should match snapshot', () => {
      expect(discardCard(mockCard)).toMatchSnapshot();
    });
  });

  describe('discardHand', () => {
    it('Should return an action with correct type', () => {
      const expected = { type: 'DISCARD_HAND' };
      const result = discardHand();
      expect(result).toEqual(expected);
    });
  });

  describe('hideCard', () => {
    const mockCard = {
      id: 1,
      name: 'Ace of Spades',
    };
    it('Should return an action with correct type', () => {
      const expected = 'HIDE_CARD';
      const result = hideCard(mockCard);
      expect(result.type).toBe(expected);
    });
    it('Should return an action with passed card', () => {
      const result = hideCard(mockCard);
      expect(result.card).toEqual(mockCard);
    });
    it('Should match snapshot', () => {
      expect(hideCard(mockCard)).toMatchSnapshot();
    });
  });

  describe('revealCard', () => {
    const mockCard = {
      id: 1,
      name: 'Ace of Spades',
    };
    it('Should return an action with correct type', () => {
      const expected = 'REVEAL_CARD';
      const result = revealCard(mockCard);
      expect(result.type).toBe(expected);
    });
    it('Should return an action with passed card', () => {
      const result = revealCard(mockCard);
      expect(result.card).toEqual(mockCard);
    });
    it('Should match snapshot', () => {
      expect(revealCard(mockCard)).toMatchSnapshot();
    });
  });
});
