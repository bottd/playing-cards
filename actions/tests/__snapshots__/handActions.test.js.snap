// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hand actions Action types Should match snapshot 1`] = `
Object {
  "addCardToHand": "ADD_CARD_TO_HAND",
  "discardCard": "DISCARD_CARD",
  "discardHand": "DISCARD_HAND",
  "hideCard": "HIDE_CARD",
  "revealCard": "REVEAL_CARD",
}
`;

exports[`Hand actions addCardToHand Should match snapshot 1`] = `
Object {
  "card": Object {
    "id": 1,
    "name": "Ace of Spades",
  },
  "type": "ADD_CARD_TO_HAND",
}
`;

exports[`Hand actions discardCard Should match snapshot 1`] = `
Object {
  "card": Object {
    "id": 1,
    "name": "Ace of Spades",
  },
  "type": "DISCARD_CARD",
}
`;

exports[`Hand actions hideCard Should match snapshot 1`] = `
Object {
  "card": Object {
    "id": 1,
    "name": "Ace of Spades",
  },
  "type": "HIDE_CARD",
}
`;

exports[`Hand actions revealCard Should match snapshot 1`] = `
Object {
  "card": Object {
    "id": 1,
    "name": "Ace of Spades",
  },
  "type": "REVEAL_CARD",
}
`;
